lib/containers/common.pm

```diff
@@ -193,25 +210,36 @@ sub test_container_image {
     die 'Argument $image not provided!'   unless $image;
     die 'Argument $runtime not provided!' unless $runtime;
 
-    # Pull the image if necessary
-    if (script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'") != 0) {
-        assert_script_run("$runtime pull $image", timeout => 300);
-        assert_script_run("$runtime image inspect --format='{{.RepoTags}}' $image | grep '$image'");
-    }
-
     my $smoketest = "/bin/uname -r; /bin/echo \"Heartbeat from $image\"";
-    assert_script_run("$runtime container create --name 'testing' '$image' /bin/sh -c '$smoketest'");
-    assert_script_run("$runtime container start 'testing'");
-    assert_script_run("$runtime wait 'testing'", 90);
-    assert_script_run("$runtime container logs 'testing' | tee '$logfile'");
-    assert_script_run("$runtime container rm 'testing'");
-    if (script_run("grep \"`uname -r`\" '$logfile'") != 0) {
-        upload_logs("$logfile");
-        die "Kernel smoke test failed for $image";
+
+    if ($runtime =~ /buildah/) {
+        if (script_run("buildah images | grep '$image'") != 0) {
+            assert_script_run("buildah pull $image", timeout => 300);
+            assert_script_run("buildah inspect --format='{{.FromImage}}' $image | grep '$image'");
+        }
+        my $container = script_output("echo \$(buildah from $image)");
```

Comment:
Why this `echo`?

Reply:
this should return the name of the container.

Reply:
but it isnt needed it. so i removed it